job.go

```diff
@@ -44,11 +59,19 @@ func NewJob(interval uint64) *Job {
 
 // Run the Job and immediately reschedule it
 func (j *Job) run() {
-	callJobFuncWithParams(j.funcs[j.jobFunc], j.fparams[j.jobFunc])
-	j.runCount++
+	switch j.runConfig.mode {
+	case SingletonMode:
+		_, j.err, _ = j.limiter.Do("main", func() (interface{}, error) {
```

Comment:
Just asking for the sake of it, but is this singleton mode related to the specific job in question? Because the way it is written here it will affect other singleton mode jobs as well, as they will all await for the "main"  keyword. If that's not expected, we could change this key to something job-specific in order to identify each particular job.

Reply:
Also, I'm sorry for taking so long to respond, I've been super busy nowadays :( will try to check in here more often

Reply:
Each job has its own ```limiter singleflight.Group```

```go
func (g *Group) Do(key string, fn func() (interface{}, error)) (v interface{}, err error, shared bool) {
	g.mu.Lock()
	if g.m == nil {
		g.m = make(map[string]*call)
	}
        // here
	if c, ok := g.m[key]; ok {
```